src/http/client.cr

```diff
@@ -105,21 +105,21 @@ class HTTP::Client
   # ```
   {% if flag?(:without_openssl) %}
     getter! tls : Nil
-    @socket : TCPSocket | Nil
     alias TLSContext = Bool | Nil
   {% else %}
     getter! tls : OpenSSL::SSL::Context::Client
-    @socket : TCPSocket | OpenSSL::SSL::Socket | Nil
     alias TLSContext = OpenSSL::SSL::Context::Client | Bool | Nil
   {% end %}
 
   # Whether automatic compression/decompression is enabled.
   property? compress : Bool = true
 
+  @socket : IO?
```

Comment:
This might seem weird but if this is no longer a socket and more generic should this variable name be updated.
I think that could make introduction to this code and working on it easier is also more accurate to name this `io` or something similar.